recipes/skyr-url/all/conanfile.py

```diff
@@ -0,0 +1,103 @@
+        return {
+            "Visual Studio": "16",
+            "gcc": "7",
+            "clang": "5",
+            "apple-clang": "10",
+        }
+
+    def config_options(self):
+        if self.settings.os == 'Windows':
+            del self.options.fPIC
+
+    def configure(self):
+        if self.settings.get_safe("compiler.cppstd"):
+            tools.check_min_cppstd(self, self._minimum_cpp_standard)
+        min_version = self._minimum_compilers_version.get(
+            str(self.settings.compiler))
+        if not min_version:
+            self.output.warn("{} recipe lacks information about the {} compiler support.".format(
+                self.name, self.settings.compiler))
+        else:
+            if tools.Version(self.settings.compiler.version) < min_version:
+                raise ConanInvalidConfiguration("{} requires C++17 support. The current compiler {} {} does not support it.".format(
+                    self.name, self.settings.compiler, self.settings.compiler.version))
+
+    def requirements(self):
+        self.requires("tl-expected/1.0.0")
+        self.requires("range-v3/0.11.0")
+        if self.options.with_json:
+            self.requires("nlohmann_json/3.9.0")
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        extracted_dir = "url-" + self.version
+        os.rename(extracted_dir, self._source_subfolder)
+
+    def _configure_cmake(self):
+        if self._cmake:
+            return self._cmake
+
+        self._cmake = CMake(self)
+        self._cmake.definitions["skyr_BUILD_TESTS"] = "OFF"
+        self._cmake.definitions["skyr_FULL_WARNINGS"] = "OFF"
+        self._cmake.definitions["skyr_WARNINGS_AS_ERRORS"] = "OFF"
+        self._cmake.definitions["skyr_USE_STATIC_CRT"] = not self.options.shared
+        self._cmake.definitions["skyr_ENABLE_JSON_FUNCTIONS"] = not self.options.with_json
+        self._cmake.configure(build_folder=self._build_subfolder)
+        return self._cmake
+
+    def build(self):
+        cmake = self._configure_cmake()
+        cmake.build()
+
+    def package(self):
+        self.copy("LICENSE_1_0.txt", src=self._source_subfolder, dst="licenses")
+        cmake = self._configure_cmake()
+        cmake.install()
+        tools.rmdir(os.path.join(self.package_folder, "share"))
+
+    def package_info(self):
+        self.cpp_info.filenames["cmake_find_package"] = "skyr-url"
+        self.cpp_info.filenames["cmake_find_package_multi"] = "skyr-url"
+        self.cpp_info.names["cmake_find_package"] = "skyr"
+        self.cpp_info.names["cmake_find_package_multi"] = "skyr"
+        self.cpp_info.components["url"].name = "skyr-url"
+        self.cpp_info.components["url"].libs = self.collect_libs()
+        self.cpp_info.components["url"].requires = ["tl-expected::tl-expected", "range-v3::range-v3" ]
+        if self.options.with_json:
+            self.cpp_info.components["url"].requires.append("nlohmann_json::nlohmann_json")
```

Comment:
```suggestion
        self.cpp_info.name = "skyr-url"
        self.cpp_info.libs = self.collect_libs()
```
Don't require components, as it's a single component.

Reply:
but this does looking like it will produce the `skyr::skyr-url` CMake target https://github.com/cpp-netlib/url#build-script